cpp/include/cudf/wrappers/timestamps.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <simt/chrono>
+
+/**---------------------------------------------------------------------------*
+ * @file timestamps.hpp
+ * @brief Concrete type definitions for int32_t and int64_t timestamps in
+ * varying resolutions as durations since the UNIX epoch.
+ *---------------------------------------------------------------------------**/
+namespace cudf {
+
+namespace detail {
+
+    // TODO: Use chrono::utc_clock when available in libcu++?
+    template <class Duration>
+    using time_point = simt::std::chrono::time_point<simt::std::chrono::steady_clock, Duration>;
+
+    template <class Duration>
+    struct timestamp : time_point<Duration> {
+        /// Initialize timestamp to 1/1/1970:00:00:00
+        constexpr timestamp() : time_point<Duration>(Duration(0)) {};
+        constexpr timestamp(Duration d) : time_point<Duration>(d) {};
+        constexpr timestamp(typename Duration::rep t) : time_point<Duration>(Duration(t)) {};
+    };
```

Comment:
For sanity, can we add something like this?
```
static_assert(sizeof(timestamp) == sizeof(Duration::rep),"");
```

Reply:
Done in https://github.com/rapidsai/cudf/pull/2918/commits/aa2372cd9baa509a6af826ded85bcf74ffbef517#diff-777506cef1eaa73782fd6985a6b0f430R69